.travis.yml

```diff
@@ -107,41 +172,52 @@ before_install:
               SOS_ENABLE_ERROR_TESTS=0
           else
               echo "Cannot test Intel compiler on a fork.  Exiting now."
-              exit 0
+              travis_terminate 0
           fi
       fi
-    ## Build libev
-    - cd $TRAVIS_SRC
-    - wget http://dist.schmorp.de/libev/Attic/libev-4.22.tar.gz
-    - tar -xzvf libev-4.22.tar.gz
-    - cd libev-4.22
-    - ./configure --prefix=$TRAVIS_INSTALL/libev
-    - make install
     ## Build Portals 4
-    - cd $TRAVIS_SRC
-    - git clone --depth 10 https://github.com/regrant/portals4.git portals4
-    - cd portals4
-    - ./autogen.sh
-    - ./configure --prefix=$TRAVIS_INSTALL/portals4/ --with-ev=$TRAVIS_INSTALL/libev --enable-zero-mrs --enable-reliable-udp --disable-pmi-from-portals
-    # JSD: --enable-transport-shmem removed; it was causing tests to hang
-    - make $TRAVIS_PAR_MAKE
-    - make install
+    - >
+      if [[ $SOS_TRANSPORT_OPTS = *"with-portals4"* ]]; then
+        cd $TRAVIS_SRC
+        git clone --depth 10 https://github.com/regrant/portals4.git portals4
+        cd portals4
+        ./autogen.sh
+        ./configure --prefix=$TRAVIS_INSTALL/portals4/ --enable-zero-mrs --enable-reliable-udp --disable-pmi-from-portals
+        # JSD: --enable-transport-shmem removed; it was causing tests to hang
+        make $TRAVIS_PAR_MAKE
+        make install
+      fi
     ## Build libfabric
-    - cd $TRAVIS_SRC
-    - git clone -b v1.6.x --depth 10 https://github.com/ofiwg/libfabric.git libfabric
-    - cd libfabric
-    - ./autogen.sh
-    - ./configure --prefix=$TRAVIS_INSTALL/libfabric
-    - make $TRAVIS_PAR_MAKE
-    - make install
-    ## Build Hydra
-    - cd $TRAVIS_SRC
-    - wget http://www.mpich.org/static/downloads/3.2/hydra-3.2.tar.gz
-    - tar xvzf hydra-3.2.tar.gz
-    - cd hydra-3.2/
-    - ./configure --prefix=$TRAVIS_INSTALL/hydra
-    - make $TRAVIS_PAR_MAKE
-    - make install
+    - >
+      if [[ $SOS_TRANSPORT_OPTS = *"with-ofi"* ]]; then
+        cd $TRAVIS_SRC
+        git clone -b v1.6.x --depth 10 https://github.com/ofiwg/libfabric.git libfabric
+        cd libfabric
+        ./autogen.sh
+        ./configure --prefix=$TRAVIS_INSTALL/libfabric
+        make $TRAVIS_PAR_MAKE
+        make install
+      fi
+    ## Build XPMEM
+    - >
+      if [[ $SOS_BUILD_OPTS = *"with-xpmem"* ]]; then
+        ## Intall Linux headers (may not be same as linux-gcp package, so has to be scripted)
+        uname -a
+        sudo apt-get install linux-headers-`uname -r`
+        ## Install XPMEM
+        cd $TRAVIS_SRC
+        git clone --depth 10 https://github.com/hjelmn/xpmem.git
+        cd xpmem
+        ./autogen.sh
+        ./configure --prefix=$TRAVIS_INSTALL/xpmem
+        make $TRAVIS_PAR_MAKE
+        make install
+        ## Hack to get SOS to see the XPMEM library
+        ln -s $TRAVIS_INSTALL/xpmem/lib $TRAVIS_INSTALL/xpmem/lib64
```

Comment:
I wonder what the cause of this is... I swear I've had to do this before, but I cannot remember which library it was.  Is it a libtool bug?

Reply:
SOS configury looks for the library in the $WITH_XPMEM/lib64.  🤦‍♂️

Reply:
Ref #685

Reply:
Haha!  I thought I'd seen this before.  Maybe that library was actually xpmem...  😖